src/views/studio/studio-comments.jsx

```diff
@@ -30,12 +31,22 @@ const StudioComments = ({
     canRestoreComment,
     handleDeleteComment,
     handleRestoreComment,
+    handleResetComments,
     handleReportComment,
     handleLoadMoreReplies
 }) => {
     useEffect(() => {
         if (comments.length === 0) handleLoadMoreComments();
-    }, []); // Only runs once after the first render
+    }, [comments.length === 0]);
+
+    // The comments you see depend on your admin status
+    // so reset them if isAdmin changes.
+    const adminRef = useRef(isAdmin);
+    useEffect(() => {
+        const wasAdmin = adminRef.current;
+        adminRef.current = isAdmin;
+        if (isAdmin !== wasAdmin) handleResetComments();
+    }, [isAdmin]);
```

Comment:
To confirm I'm reading this right...
Does this mean that handleResetComments will run if we go from not being an admin -> being an admin, or will it also run if we go from being an admin -> not being an admin?

Reply:
Actually I think the test answers my question